src/api/submission/writer.py

```diff
@@ -69,13 +70,18 @@ def validate_data(self, serializer_cls, data):
 
         return serializer
 
-    def get_or_create(self, model, serializer_cls, data):
-        data["id"] = data.get("id") or uuid.uuid4()
+    def get_or_create(self, model, serializer_cls, data, additional_data=None):
+        pk = data.get("id") or uuid.uuid4()
+        data["id"] = pk
         serializer = self.validate_data(serializer_cls, data)
         try:
             data.pop("id")
             return model.objects.get(**data)
         except model.DoesNotExist:
+            if isinstance(additional_data, dict):
+                data["id"] = pk
```

Comment:
Redundant, no?

Reply:
No b/c it gets stripped above the except